src/analysis/souffle/taint.dl

```diff
@@ -38,13 +38,17 @@
 // There is a claim that `accessPath` definitely has `tag` taint.
 .decl claimHasTag(accessPath: AccessPath, tag: Tag)
 
+// There is a claim that `accessPath` definitely does not have `tag` taint
+// (even if the taint analysis would otherwise have this taint).
+.decl claimNotHasTag(accessPath: AccessPath, tag: Tag)
+
 //-----------------------------------------------------------------------------
 // Rules
 //-----------------------------------------------------------------------------
 mayHaveTag(tgt, tag) :- claimHasTag(tgt, tag).
 mayHaveTag(tgt, tag) :-
     edge(src, tgt), mayHaveTag(src, tag),
-    !downgrades(tgt, tag).
+    !claimNotHasTag(tgt, tag).
```

Comment:
Again, this is likely in conflict with changes introduced by #121 .

Reply:
fixed.